enterprise/management/commands/seed_enterprise_devstack_data.py

```diff
@@ -0,0 +1,252 @@
+)
+from enterprise.models import (
+    EnterpriseCustomer,
+    EnterpriseCustomerCatalog,
+    EnterpriseCustomerUser,
+    EnterpriseFeatureRole,
+    EnterpriseFeatureUserRoleAssignment,
+    SystemWideEnterpriseRole,
+    SystemWideEnterpriseUserRoleAssignment,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+CATALOG_CONTENT_FILTER = {
+    'content_type': 'courserun',
+}
+
+class Command(BaseCommand):
+    """
+    Management command for populating Devstack with initial data for enterprise.
+
+    Example usage:
+        $ ./manage.py lms seed_enterprise_devstack_data
+    """
+    help = '''
+    Seeds an enterprise customer, users of various roles and permissions initial 
+    data in devstack for related Enterprise models.
+    '''
+
+    def _get_default_site(self):
+        """ Gets or creates the default devstack site example.com """
+        site = Site.objects.get_or_create(name='example.com')
+        return site
+
+    def _create_enterprise_customer(self, site):
+        """ Gets or creates an EnterpriseCustomer """
+        customer_name = 'Test Enterprise'
+        enterprise_customer, __ = EnterpriseCustomer.objects.get_or_create(  # pylint: disable=no-member
+            name=customer_name,
+            site_id=site.id,
+            country='US',
+            slug=slugify(customer_name),
+            enable_data_sharing_consent=True,
+            enable_portal_code_management_screen=True,
+            enable_portal_reporting_config_screen=True,
+        )
+        return enterprise_customer
+
+    def _create_catalog_for_enterprise(self, enterprise_customer):
+        """ Gets or creates a catalog for an EnterpriseCustomer """
+        catalog, __ = EnterpriseCustomerCatalog.objects.get_or_create(
+            title='All Course Runs',
+            enterprise_customer=enterprise_customer,
+            content_filter=CATALOG_CONTENT_FILTER,
+        )
+        return catalog
+
+    def _create_enterprise_data_api_group(self):
+        """ Ensures the `ENTERPRISE_DATA_API_ACCESS_GROUP` is created """
+        Group.objects.get_or_create(name=ENTERPRISE_DATA_API_ACCESS_GROUP)
+
+    def _create_enterprise_enrollment_api_group(self):
+        """ Ensures the `ENTERPRISE_ENROLLMENT_API_ACCESS_GROUP` is created """
+        Group.objects.get_or_create(name=ENTERPRISE_ENROLLMENT_API_ACCESS_GROUP)
+
+    def _create_enterprise_user(self, username, role):
+        """ 
+        Creates a new user with the specified `username` and `role` (e.g., 
+        'enterprise_learner'). The newly created user is added to the 
+        appropriate Django groups (e.g., data api access) and creates 
+        system-wide and feature role assignments.
+        """
+        valid_roles = [
+            ENTERPRISE_LEARNER_ROLE,
+            ENTERPRISE_ADMIN_ROLE,
+            ENTERPRISE_OPERATOR_ROLE,
+        ]
+        if role in valid_roles:
+            is_staff = True if role == ENTERPRISE_OPERATOR_ROLE else False
+            try:
```

Comment:
Why not user `get_or_create` here?

Reply:
I was originally using `get_or_create` for creating a user here, but then the user's password was not properly being set/hashed, preventing the newly created user from being able to sign into the account.

This `create_user` method handles setting the user's password correctly.

See https://stackoverflow.com/a/23482284/9578634.

Reply:
Oh interesting, didn't know that. Good find!